### ISSUE TITLE
Flag imports from importing modules

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#12194]
say i have modules `a.py`, `b.py`, `c.py` that each define a class `A`, `B` and `C`.

`b.py` looks like this

```py
from pkg.a import A

class B(A):
    pass
```

`c.py` can then import `A` from `b` without raising a lint error.

```py
from pkg.b import A, B

class C(A, B):
    pass
```

would be nice if `ruff` had a rule to enforce that symbols are imported from the defining module, not another importing module:

```py
from pkg.b import A, B # bad

# good
from pkg.a import A
from pkg.b import B

class C(A, B):
    pass
```
